### PR TITLE
Update ESLint

### PR DESCRIPTION
**Pull request by desfero**
**Technical notes**
ESLint did a really good progress recently with typescript linting support. After the update we got more than 300 linting errors related to invalid types/any's (a couple of them caught a real bug in the code). 

Not all violations were fixed, a lot of any related violations in spec were just marked as ignored given its quite time consuming to fix all of them at once.

### CODE REVIEW

File: packages/wallet/.eslintignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-node_modules
+.eslintrc.js
```
Comment (arjunu) [inline packages/wallet/.eslintignore:1]: it doesn't ignore itself by default?

File: packages/shared-utils/src/utils/types.ts [hunk 2]
```diff
@@ -192,3 +192,11 @@ export type UnionToIntersection<U> = (U extends any
...
   ? I
   : never;
+
+/**
+ * Specifies an object with unknown keys
+ *
+ * Useful when you don't care about exact props passed to the component.
+ * @note Don't use `{}` as a type. `{}` actually means "any non-nullish value".
+ */
+export type UnknownObject = Record<string, unknown>;
```
Comment (arjunu) [inline packages/shared-utils/src/utils/types.ts:202]: 👍

File: packages/wallet/app/components/shared/forms/fields/Form.tsx [hunk 3]
```diff
@@ -22,7 +25,7 @@ const Form = <Values extends {}>({
     enableReinitialize
     validateOnMount
   >
-    {formikProps => (typeof children === "function" ? children(formikProps) : children)}
+    {formikProps => (isFunction(children) ? children(formikProps) : children)}
```
Comment (arjunu) [inline packages/wallet/app/components/shared/forms/fields/Form.tsx:28]: Formik uses the same :D https://github.com/formik/formik/blob/d2637207791a74489c5d57532a0c23dbf0a70b22/packages/formik/src/utils.ts#L12

File: packages/wallet/app/modules/eth/lib/EthSecureEnclave.ts [hunk 4]
```diff
@@ -192,7 +192,7 @@ class EthSecureEnclave {
     try {
       const hdNode = utils.HDNode.fromMnemonic(secret).derivePath(derivationPath);
       return this.addSecret(hdNode.privateKey);
-    } catch (e) {
```
Comment (arjunu) [inline packages/wallet/app/modules/eth/lib/EthSecureEnclave.ts:195]: don't we want the original exception in the message?

File: packages/wallet/app/modules/wallet-connect/lib/WalletConnectAdapter.ts [hunk 5]
```diff
@@ -108,6 +114,8 @@ class WalletConnectAdapter extends EventEmitter2 {
         result,
       });
     } catch (error) {
+      assertError(error);
```
Comment (arjunu) [inline packages/wallet/app/modules/wallet-connect/lib/WalletConnectAdapter.ts:117]: when would this not be an instance of error?

File: packages/wallet/app/modules/eth/lib/EthSecureEnclave.ts (hunk 4 above)
Comment (arjunu) [reply packages/wallet/app/modules/eth/lib/EthSecureEnclave.ts:195]: same question for other places

File: packages/wallet/.eslintignore (hunk 1 above)
Comment (desfero) [reply packages/wallet/.eslintignore:1]: that just changed in a recent update

File: packages/wallet/app/modules/wallet-connect/lib/WalletConnectAdapter.ts (hunk 5 above)
Comment (desfero) [reply packages/wallet/app/modules/wallet-connect/lib/WalletConnectAdapter.ts:117]: i would say it depends. You can `throw` anything you want, from number/string to even Promise (that's btw exactly how react suspense is implemented)

File: packages/wallet/app/modules/eth/lib/EthSecureEnclave.ts (hunk 4 above)
Comment (desfero) [reply packages/wallet/app/modules/eth/lib/EthSecureEnclave.ts:195]: good point, fixed

**Pull request merged**